##### Algumas funções simples: avaliação do IMC (em inglês, BMI) e conversão de unidades imperiais em unidades métricas
    
    Some simple functions: evaluating BMI and converting imperial units to metric units

Veja o código no editor. Há duas coisas a que temos de prestar atenção.
    
    Look at the code in the editor. There are two things we need to pay attention to.

In [1]:
def bmi(weight, height):
    if height < 1.0 or height > 2.5 or \
    weight < 20 or weight > 200:
        return None

    return weight / height ** 2


print(bmi(352.5, 1.65))


None


Primeiro, a invocação de teste assegura que a proteção funciona corretamente - o output é:
    
    First, the test invocation ensures that the protection works properly - the output is:

None

Em segundo lugar, dê uma vista de olhos na forma como o símbolo da barra invertida (\) é usado. Se o utilizar em código Python e terminar uma linha com ele, dirá ao Python para continuar a linha de código na próxima linha de código.
    
    Second, take a look at the way the backslash (\) symbol is used. If you use it in Python code and end a line with it, it will tell Python to continue the line of code in the next line of code.

Pode ser particularmente útil quando se tem de lidar com longas linhas de código e se gostaria de melhorar a legibilidade do código.
    
    It can be particularly useful when you have to deal with long lines of code and you'd like to improve code readability.

Está bem, mas há algo que omitimos com demasiada facilidade - as medidas imperiais. Esta função não é muito útil para pessoas acostumadas a libras, pés e polegadas.
    
    Okay, but there's something we omitted too easily - the imperial measurements. This function is not too useful for people accustomed to pounds, feet and inches.

O que pode ser feito por elas?
    
    What can be done for them?

Podemos escrever duas funções simples para converter unidades imperiais em unidades métricas. Vamos começar com as libras.
    
    We can write two simple functions to convert imperial units to metric ones. Let's start with pounds.
    
É um facto bem conhecido que 1 lb = 0.45359237 kg. Utilizaremos isto na nossa nova função.
    
    It is a well-known fact that 1 lb = 0.45359237 kg. We'll use this in our new function.

Esta é a nossa função auxiliar, chamada lb_to_kg:
    
    This is our helper function, named lb_to_kg:

In [2]:
def lb_to_kg(lb):
    return lb * 0.45359237
print(lb_to_kg(1))

0.45359237


E agora é altura para os pés e polegadas: 1 ft = 0.3048 m, e 1 in = 2.54 cm = 0.0254 m.
    
    And now it's time for feet and inches: 1 ft = 0.3048 m, and 1 in = 2.54 cm = 0.0254 m.

A função que escrevemos é chamada ft_and_inch_to_m:
    
    The function we've written is named ft_and_inch_to_m:

In [3]:
def ft_and_inch_to_m(ft, inch):
    return ft * 0.3048 + inch * 0.0254
print(ft_and_inch_to_m(1, 1))

0.3302


Parece como o esperado.
    
    It looks as expected.

Nota: queríamos nomear o segundo parâmetro apenas in, não inch, mas não podíamos. Sabe porquê?
    
    Note: we wanted to name the second parameter just in, not inch, but we couldn't. Do you know why?
    
in é uma keyword Python - não pode ser usada como um nome.
    
    in is a Python keyword - it cannot be used as a name.

Vamos converter seis pés em metros:
    
    Let's convert six feet into meters:

In [4]:
print(ft_and_inch_to_m(6, 0))

1.8288000000000002


É bem possível que por vezes se queira usar apenas os pés sem polegadas. O Python vai ajudá-lo? Claro que vai.
    
    It's quite possible that sometimes you may want to use just feet without inches. Will Python help you? Of course it will.

Modificamos um pouco o código:
    
    We've modified the code a bit:

In [5]:
def ft_and_inch_to_m(ft, inch = 0.0):
    return ft * 0.3048 + inch * 0.0254
print(ft_and_inch_to_m(6))

1.8288000000000002


Agora o parâmetro inch tem o seu valor padrão igual a 0.0.
    
    Now the inch parameter has its default value equal to 0.0.

O código produz o seguinte output - isto é o que é esperado:
    
    The code produces the following output - this is what is expected:

1.8288000000000002

Finalmente, o código é capaz de responder à pergunta: qual é o IMC de uma pessoa com 1,75 m de altura e pesando 176 lbs?
    
    Finally, the code is able to answer the question: what is the BMI of a person 5'7" tall and weighing 176 lbs?

Este é o código que construímos:
    
    This is the code we have built:

In [6]:
def ft_and_inch_to_m(ft, inch = 0.0):
    return ft * 0.3048 + inch * 0.0254
def lb_to_kg(lb):
    return lb * 0.45359237
def bmi(weight, height):
    if height < 1.0 or height > 2.5 or weight < 20 or weight > 200:
        return None
    return weight / height ** 2
print(bmi(weight = lb_to_kg(176), height = ft_and_inch_to_m(5, 7)))

27.565214082533313


Execute o código e teste-o.

    Run the code and test it.